extensions/zmh-program/chatgpt-en.js

```diff
@@ -0,0 +1,73 @@
+(function(Scratch) {
+  'use strict';
+  let apikey, cache = {};
+
+  class ChatGPT {
+    getInfo () {
+      return {
+        id: 'openai',
+        name: 'ChatGPT',
+        color1: '#0e0e0e',
+        blocks: [
+          {
+            opcode: 'ask',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'ask chatgpt [text]',
+            arguments: {
+              text: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: 'hi'
+              }
+            },
+            disableMonitor: true,
+          },
+          {
+            opcode: 'customize',
+            blockType: Scratch.BlockType.COMMAND,
+            text: 'customize the apikey [key]  (risky)',
+            arguments: {
+              key: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: 'sk-'
+              }
+            },
+          },
+          {
+            opcode: 'donate',
+            blockType: Scratch.BlockType.COMMAND,
+            text: 'donate'
+          },
+        ]
+      };
+    }
+    async ask ({ text }) {
+      const content = text.trim();
+      if (!content) return 'Please enter the content!';
+      if (cache[text]) return cache[text];
+      try {
+        // eslint-disable-next-line no-restricted-syntax
+        const resp = await fetch('https://chatgpt.deeptrain.net/gpt', {
+          method: 'POST',
+          headers: { 'Content-Type': 'application/json' },
+          body: JSON.stringify({ data: content, key: apikey })
```

Comment:
if someone provides their own API key then it must go directly to OpenAI so you can't be logging them on your server

Reply:
we also don't want people to store API keys inside projects so they don't accidentally get credit card charges; the extension needs some custom interface to get the key other than a Scratch block (tip: we have buttons now - https://github.com/TurboWarp/extensions/blob/master/website/test/button.js)

Reply:
Thanks! Should I add an if? OK.And I'm trying to set button.
But you know that openai cannot be logged in China Mainland. That's why I added it to the proxy server.
But you are also right that sending apikey to a third-party server is not a good idea.